apps/postgresql-server/bin/postgresql.sh

```diff
@@ -19,5 +19,8 @@ else
     echo "Done applying schema migrations."
 fi
 
+# dump schema file for reference in development (run ./dev/get_schema.sh to get local copy)
+pg_dump mediacloud > /opt/postgresql-server/schema/mediawords.sql
+
```

Comment:
This here has to be done once, at Docker build time, not in the PostgreSQL service startup wrapper, otherwise it will attempt to build the whole 16 TB worth of data on every PostgreSQL start :)

Reply:
Back to `initialize_db.sh` it goes 🙂 

This is the result of me throwing the kitchen sink at this PR because of a funny problem wherein the GitHub Actions machine for some reason fails to create the `postgresql-server/schema` directory, despite successfully creating the other subdirectories specified in [the same `Dockerfile` command](https://github.com/mediacloud/backend/pull/759/files#diff-97559d4961873beb9ff7b69e99dbaff753262a2a30deb9790dbe8344689000c7R33-R37) (`bin`, `migrations`, `pgmigrate-callbacks`). Illustrated here via [tmate debugging](https://github.com/mediacloud/backend/pull/759/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R193-R197) (i.e. `ssh`ing to the box running the GHA build):

```
runner@fv-az118-420:~/work/backend/backend/apps/postgresql-server$ ls
Dockerfile  bin  migrations  migrations.yml  pgmigrate-callbacks
```

Consequently, the GHA builds are [failing](https://github.com/mediacloud/backend/pull/759/checks?check_run_id=2880302777) with the error `Cannot create container for service postgresql-server: invalid mount config for type "bind": bind source path does not exist: /home/runner/work/backend/backend/apps/postgresql-server/schema`.

When I run the build and create the container locally, I face no such problem and the tests pass:

```
postgres@b2be31a6edca:/opt$ cd postgresql-server
postgres@b2be31a6edca:/opt/postgresql-server$ ls
bin  migrations  migrations.yml  pgmigrate-callbacks  schema
postgres@b2be31a6edca:/opt/postgresql-server$ cd schema
postgres@b2be31a6edca:/opt/postgresql-server/schema$ ls
mediawords.sql
```
Do you have any hypotheses about the cause of this disconnect between build contexts?

Reply:
In `docker-compose.tests.yml` we try to mount as source code from the host to the container, on top of what got baked in via `Dockerfile`; the idea is that if the source code / configuration / whatever gets changed on the host (e.g. when you're actively developing things and trying to make something work), you wouldn't have to rebuild images of affected apps after each and every change.

We do this not only for the "app" at hand but for its dependencies too, e.g. one can write a migration and run a test straight away without having to rebuild `postgresql-server` (because the migration would get applied on the container start).

Therefore, in `docker-compose.tests.yml` we mount the `apps/postgresql-server/bin/`, `apps/postgresql-server/schema/` (which used to include `migrations/`), and `apps/postgresql-base/conf/`:

https://github.com/mediacloud/backend/blob/133ae634d7809c8e6fdbcb9931d335d2b4420701/apps/common/docker-compose.tests.yml#L86-L101

I think the problem that you're encountering here is that the `schema/` directory (rightfully) gets removed on this branch, and Git doesn't store information about directories (only about files):

```bash
$ git status
On branch jot-pgmigrate
Your branch is up to date with 'origin/jot-pgmigrate'.
nothing to commit, working tree clean

$ mkdir git_doesnt_care_about_directories

$ git status                             
On branch jot-pgmigrate
Your branch is up to date with 'origin/jot-pgmigrate'.
nothing to commit, working tree clean
```

so while an empty `schema/` directory exists on your laptop, `schema/` is nowhere to be found right after checking out the repository on the GitHub actions worker. Then `docker-compose.tests.yml` of various apps try to mount that `schema/` directory but it doesn't exist (and shouldn't), and so all of the tests fail. When you resurrected `apps/postgresql-server/schema/migrations.sql`, `apps/postgresql-server/schema/` started existing again, and tests started passing again :)

Given that `schema/` is now gone, you should update (search-and-replace basically) instances of `postgresql-server` in all of the `docker-compose.tests.yml`. I think that for that service it would be sensible to mount the following things from the host:

* `postgresql-server/bin/` - PostgreSQL server startup wrapper script
* `postgresql-server/migrations/` - new migrations location
* `postgresql-base/conf/` - PostgreSQL configuration

Also get rid of `postgresql-server/schema/mediawords.sql` from the source tree, it's better off in root (`/`).

It's not applicable here, but in the future if you'd like to create an empty ignored directory in a Git repository, one hack to do it is to add a `.gitignore` like this one:

https://github.com/mediacloud/backend/blob/133ae634d7809c8e6fdbcb9931d335d2b4420701/apps/solr-base/src/solr/collections/mediacloud/data/.gitignore